File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = defaultdict(int)
        for i in nums:
            mp[i]+=1
        ls = [[] for _ in range(len(nums)+1)]
        for key,val in mp.items():
            ls[val].append(key)
        out =[]
        for i in range (len(nums),-1,-1):
            if len(ls[i])>0:
                while k>0 and len(ls[i])>0:
                    out.append(ls[i][-1])
                    ls[i].pop()
                    k-=1
            if k==0:
                return out

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] out  =new int[nums.length];
        int pref = 1;
        for(int i = 0 ; i<nums.length; i++ ){
            out[i] = pref;
            pref *= nums[i];
        }
        int post = 1;
        for(int i = nums.length -1  ; i>=0; i-- ){
            out[i] *= post;
            post *= nums[i];
        }

        return out;
    }
}  
